File: data/src/main/java/mega/privacy/android/data/model/chat/NonContactInfo.java
package mega.privacy.android.data.model.chat;


public class NonContactInfo {

    String handle;
    String fullName;
    String firstName;
    String lastName;
    String email;
//    boolean requestDone = false;

    public NonContactInfo(String handle, String fullName, String firstName, String lastName, String email) {
        this.fullName = fullName;
        this.handle = handle;
        this.firstName = firstName;
        this.lastName = lastName;
        this.email = email;
//        this.requestDone = requestDone;
    }

    public String getFullName() {

        if(firstName==null){
            firstName="";
        }
        if(lastName == null){
            lastName="";
        }

        if (firstName.trim().length() <= 0){
            fullName = lastName;
        }
        else{
            fullName = firstName + " " + lastName;
        }

        return fullName;
    }

    public void setFullName(String fullName) {
        this.fullName = fullName;
    }

    public String getHandle() {
        return handle;
    }

    public void setHandle(String handle) {
        this.handle = handle;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }
}

File: data/src/main/java/mega/privacy/android/data/model/chat/NonContactInfo.kt
package mega.privacy.android.data.model.chat

/**
 * Data class to store information about a non-contact.
 *
 * @property handle    Non-contact handle.
 * @property fullName  Non-contact full name.
 * @property firstName Non-contact first name.
 * @property lastName  Non-contact last name.
 * @property email     Non-contact email.
 */
data class NonContactInfo(
    val handle: String,
    val fullName: String?,
    val firstName: String?,
    val lastName: String?,
    val email: String?,
) {
    /**
     * Short name of the non-contact.
     */
    val shortName: String?
        get() = firstName?.takeIf { it.isNotBlank() }
            ?: lastName?.takeIf { it.isNotBlank() }
            ?: email
}